LLama/LLamaTemplate.cs

```diff
@@ -278,10 +291,29 @@ unsafe int ApplyInternal(Span<LLamaChatMessage> messages, byte[] output)
         }
     }
 
+    /// <summary>
+    /// Apply the template to the messages and return the resulting prompt as a string
+    /// </summary>
+    /// 
+    /// <returns>The formatted template string as defined by the model</returns>
+    public string ToModelPrompt()
```

Comment:
I deliberately avoided creating a method like this when creating LlamaTemplate originally. Calling this method will allocate a new (and potentially very large, since it contains the entire chat history) string every time it is called.

All the downstream uses of the template (e.g. tokenizing) should be built to accept a `Span<byte>` or `Span<char>`, so in principle this should never be needed. What's this used for?

Reply:
(sidenote: if we decide to keep it, it's probably better to call it `ToString()`)

Reply:
I see below this is used in `PromptTemplateTransformer`. Could this entire method be moved there for now? That way we're not encouraging a sub-optimal use in the API (and the internals of `PromptTemplateTransformer` can be optimised if necessary later).

Reply:
Good compromise, I'll move it